src/lib.rs

```diff
@@ -0,0 +1,408 @@
+                    .collect()
+            }
+            ScopeMode::AllowList(patterns) => {
+                meta.packages
+                    .iter()
+                    // Deviating from cargo by not supporting patterns
+                    // Deviating from cargo by only checking workspace members
+                    .filter(|p| workspace_members.contains(&p.id) && patterns.contains(&p.name))
+                    .map(|p| &p.id)
+                    .collect()
+            }
+        };
+
+        meta.packages
+            .iter()
+            .filter(|&p| base_ids.contains(&p.id))
+            .collect()
+    }
+}
+
+impl Check {
+    pub fn new(current: Rustdoc) -> Self {
+        Self {
+            scope: Scope::default(),
+            current,
+            baseline: Rustdoc::from_latest_version(),
+            log_level: Default::default(),
+        }
+    }
+
+    pub fn with_package_selection(&mut self, selection: PackageSelection) -> &mut Self {
+        self.scope.mode = ScopeMode::DenyList(selection);
+        self
+    }
+
+    pub fn with_packages(&mut self, packages: Vec<String>) -> &mut Self {
+        self.scope.mode = ScopeMode::AllowList(packages);
+        self
+    }
+
+    pub fn with_baseline(&mut self, baseline: Rustdoc) -> &mut Self {
+        self.baseline = baseline;
+        self
+    }
+
+    pub fn with_log_level(&mut self, log_level: log::Level) -> &mut Self {
+        self.log_level = Some(log_level);
+        self
+    }
+
+    pub fn check_release(&self) -> anyhow::Result<Report> {
+        let mut config = GlobalConfig::new().set_level(self.log_level);
+
+        let loader: Box<dyn baseline::BaselineLoader> = match &self.baseline.source {
+            RustdocSource::Rustdoc(path) => {
+                Box::new(baseline::RustdocBaseline::new(path.to_owned()))
+            }
+            RustdocSource::Root(root) => Box::new(baseline::PathBaseline::new(root)?),
+            RustdocSource::Revision(root, rev) => {
+                let metadata = manifest_metadata_no_deps(root)?;
+                let source = metadata.workspace_root.as_std_path();
+                let slug = util::slugify(rev);
+                let target = metadata
+                    .target_directory
+                    .as_std_path()
+                    .join(util::SCOPE)
+                    .join(format!("git-{slug}"));
+                Box::new(baseline::GitBaseline::with_rev(
+                    source,
+                    &target,
+                    rev,
+                    &mut config,
+                )?)
+            }
+            RustdocSource::Version(version) => {
+                let mut registry = {
+                    let manifest_dir = match &self.current.source {
+                        RustdocSource::Root(manifest_dir)
+                        | RustdocSource::Revision(manifest_dir, _) => manifest_dir,
+                        RustdocSource::Version(_) | RustdocSource::Rustdoc(_) => {
```

Comment:
Hmm is `--current-rustdoc` not supported together with `--baseline-version`? I'm not sure why that might be the case? I'd have expected it to just assume that the manifest is in the current directory and up-to-date in that case, and bail if it is not.

If indeed it doesn't work today, then I will update my own prior — no code change needed in that case :)

Reply:
After the changes from [tonowak](https://github.com/tonowak), the `anyhow::bail` is not there anymore 👍